### ISSUE TITLE
Bug: Recompiles src files many times

### ISSUE THREAD

**Issue by pizza2code** [TypeStrong/tsify#150]
Even after paring our gulpfile down to the small example you see below, I'm getting numerous re-compiles of the same Typescript files.  If you see the console output below, `main.tsx`'s test error is reported **11 times** in total!  

This has been happening since the first time I tried tsify. I have no tsconfig.json and can't think of any other files that might interfere with the build.  RHEL 6.6, Node v5.9.0

### gulpfile.js
```javascript
var gulp = require('gulp');
var browserify = require('browserify');
var tsify = require('tsify');
var source = require('vinyl-source-stream');

var b = browserify({
    entries: ['./app/main.tsx']
});
var es6Types = 'node_modules/typescript/lib/lib.es6.d.ts';
var thirdPartyTypes = 'typings/index.d.ts';

b.add(es6Types);
b.add(thirdPartyTypes);
b.plugin(tsify, {
    jsx: "react",
    target: "ES5",
    module: "commonjs"
});

gulp.task('default', function () {
    return b.bundle()
        .on('error', function(e) { console.error(e.message ? e.message : e) })
        .pipe(source('main.js'))
        .pipe(gulp.dest('./dest'));
});
```
### running gulp
```
[22:09:38] Using gulpfile /web/pane/example/gulpfile.js
[22:09:38] Starting 'default'...
app/main.tsx(7,5): Error TS2322: Type 'number' is not assignable to type 'string'.
app/main.tsx(7,5): Error TS2322: Type 'number' is not assignable to type 'string'.
app/main.tsx(7,5): Error TS2322: Type 'number' is not assignable to type 'string'.
app/main.tsx(7,5): Error TS2322: Type 'number' is not assignable to type 'string'.
app/main.tsx(7,5): Error TS2322: Type 'number' is not assignable to type 'string'.
/web/pane/example/app/TitleBar.tsx(109,2): Error TS2354: No best common type exists among return expressions.
/web/pane/example/app/TitleBar.tsx(125,2): Error TS2354: No best common type exists among return expressions.
app/UIPane.tsx(162,2): Error TS2354: No best common type exists among return expressions.
app/main.tsx(7,5): Error TS2322: Type 'number' is not assignable to type 'string'.
/web/pane/example/app/TitleBar.tsx(109,2): Error TS2354: No best common type exists among return expressions.
/web/pane/example/app/TitleBar.tsx(125,2): Error TS2354: No best common type exists among return expressions.
app/IOPane.tsx(53,2): Error TS2354: No best common type exists among return expressions.
app/IOPane.tsx(129,2): Error TS2354: No best common type exists among return expressions.
app/UIPane.tsx(162,2): Error TS2354: No best common type exists among return expressions.
app/main.tsx(7,5): Error TS2322: Type 'number' is not assignable to type 'string'.
/web/pane/example/app/TitleBar.tsx(109,2): Error TS2354: No best common type exists among return expressions.
/web/pane/example/app/TitleBar.tsx(125,2): Error TS2354: No best common type exists among return expressions.
app/IOPane.tsx(53,2): Error TS2354: No best common type exists among return expressions.
app/IOPane.tsx(129,2): Error TS2354: No best common type exists among return expressions.
app/UIPane.tsx(162,2): Error TS2354: No best common type exists among return expressions.
app/main.tsx(7,5): Error TS2322: Type 'number' is not assignable to type 'string'.
/web/pane/example/app/TitleBar.tsx(109,2): Error TS2354: No best common type exists among return expressions.
/web/pane/example/app/TitleBar.tsx(125,2): Error TS2354: No best common type exists among return expressions.
app/IOPane.tsx(53,2): Error TS2354: No best common type exists among return expressions.
app/IOPane.tsx(129,2): Error TS2354: No best common type exists among return expressions.
app/UIPane.tsx(162,2): Error TS2354: No best common type exists among return expressions.
app/main.tsx(7,5): Error TS2322: Type 'number' is not assignable to type 'string'.
/web/pane/example/app/TitleBar.tsx(109,2): Error TS2354: No best common type exists among return expressions.
/web/pane/example/app/TitleBar.tsx(125,2): Error TS2354: No best common type exists among return expressions.
app/IOPane.tsx(53,2): Error TS2354: No best common type exists among return expressions.
app/IOPane.tsx(129,2): Error TS2354: No best common type exists among return expressions.
app/UIPane.tsx(162,2): Error TS2354: No best common type exists among return expressions.
app/main.tsx(7,5): Error TS2322: Type 'number' is not assignable to type 'string'.
[22:10:12] Finished 'default' after 34 s
```

**Comment by pizza2code** [comment]
After investigating further, it appears tsify lacks support for any sort of non-local modules, whether referenced by [NODE_PATH](https://nodejs.org/api/modules.html#modules_loading_from_the_global_folders) (using a modified Typescript) or by symlink (https://github.com/TypeStrong/tsify/issues/135).

This makes it difficult if not impossible for our org to use tsify, since we have multiple in-house JS libraries which we cannot publish nor duplicate on disk for every project.

**Comment by cartant** [comment]
This should be closed by #171. It should now be possible to use symlinks in `node_modules`.
